ckan_sdk/client.py

```diff
@@ -3,92 +3,99 @@
 import os
 from os import listdir
 from urllib.parse import urljoin
-
 import hashlib
+
 import requests
 
+from ckan_sdk import f11s
+
 log = logging.getLogger(__name__)
 
 
 class Client:
     '''
     This is a class for authentiacation and uploading the
-    data to the cloud storage
+    data to the CKAN and cloud storage
 
     Attributes:
         base_url (str): url to the server
         auth_token (str): authetication token for the request
         organization_id (str): name of the organization
-        dataset_id (str): name of the dataset
     '''
 
-    def __init__(self, base_url, auth_token, organization_id, dataset_id):
+    def __init__(self, base_url, auth_token, organization_id):
         self.base_url = base_url
         self.auth_token = auth_token
         self.organization_id = organization_id
-        self.dataset_id = dataset_id
-
-    def push(self, resource_path, on_progress=None):
-        # Given the JWT token and file size, will return signed URL, verify URL and JWT token
-
-        # Get the JWT token from CkanAuthz
-        scope = 'obj:{}/{}/*:write'.format(self.organization_id, self.dataset_id)
-        response = self.get_jwt_from_ckan_authz(scope)
-        jwt_token = response['result']['token']
-
-        file_info = self.get_file_info(resource_path)
-        lfs_response = self.request_file_upload_actions(jwt_token,
-                             file_info.get('sha256'), file_info.get('size'))
-
-        lfs_object = lfs_response.get('objects')[0]
-        result = {
-            'oid': lfs_object['oid'],
-            'size': lfs_object['size'],
-            'name': file_info.get('name'),
-            'success': True,
-            'file_already_exists': False,
-            }
 
-        # File doesn't exist in storage
-        if lfs_object['actions']:
+    def push(self, dataset, append=False):
+        # Push resources to CKAN and cloud
 
-            # Upload the file to cloud storage
-            actions = lfs_object['actions']
-            upload_url = actions['upload']['href']
-            upload_token = actions['upload']['header']
-            self.upload_to_storage(upload_url, upload_token, file_info.get('content'), on_progress)
-
-            # Verify file to cloud storage
-            verify_url = actions['verify']['href']
-            verify_token = actions['verify']['header']
-            self.verify_upload(verify_url, verify_token, file_info.get('sha256'), file_info.get('size'))
-
-            return result
+        try:
+            resources = dataset.pop('resources')
+        except KeyError:
+            log.exception('There is no resource in the dataset {}'.format(dataset.get('name')))
```

Comment:
is it an exception? it's then just a dataset with no resources - which is ok ...

I think you want to support that path as an non-exceptional option.